### DISCUSSION TITLE
Adding types to Doxygen Comments

### DISCUSSION THREAD

**Discussion by geographika** [swig/swig#1914] [Q&A] [answered]
I've been adding comments to C structs, that will be used to generate Sphinx documentation from the Python bindings.
Is there any way to automatically include the C-types in the comments without manually adding them?

For example:

```C
  struct labelObj{
    int shadowsizex; ///< See :ref:`SHADOWSIZE <mapfile-label-shadowsize>`
...
```

Bindings and doxygen comments generated using:

```
swig -python -py3 -shadow -doxygen -o mapscript_wrap.c ../mapscript.i
```

Output .py file - from which documentation is generated. Note the comments are included, but there is no type information. Ideally `int` would be included automatically somehow:

```python
class labelObj(object):
    shadowsizex = property(
        _mapscript.labelObj_shadowsizex_get, 
        _mapscript.labelObj_shadowsizex_set, 
       doc=r""" See :ref:`SHADOWSIZE <mapfile-label-shadowsize>`""")
```

**Comment by wsfulton** [comment, marked as answer]
#735 is the most relevant discussions on this topic to date and outlines what work needs to be done.

**Reply by geographika** [reply]
For reference #1951 adds type annotations, but requires Python 3.6+
